modules/activitypub/keypair_test.go

```diff
@@ -23,3 +28,40 @@ func TestKeygen(t *testing.T) {
 	assert.Regexp(t, regexp.MustCompile("^-----BEGIN PUBLIC KEY-----.*"), pub)
 
 }
+
+func TestSignUsingKeys(t *testing.T) {
+	priv, pub, err := GenerateKeyPair()
+	assert.NoError(t, err)
+
+	privPem, _ := pem.Decode([]byte(priv))
+	if privPem == nil || privPem.Type != "RSA PRIVATE KEY" {
+		t.Fatal("key is wrong type")
+	}
+
+	privParsed, err := x509.ParsePKCS1PrivateKey(privPem.Bytes)
+	if err != nil {
+		assert.NoError(t, err)
+	}
```

Comment:
Can't the if be replaced by its content?
I hope `assert.NoError` checks whether an error is present itself, otherwise that function makes only limited sense.
Correct me if this function categorically fails the test.
```suggestion
	assert.NoError(t, err)
```